Plugin/Tax/Config/ShippingTaxPlugin.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FireGento\MageSetup\Plugin\Tax\Config;
+
+use FireGento\MageSetup\Model\Config as FireGentoConfig;
+use Magento\Checkout\Model\Cart;
+use Magento\Customer\Model\ResourceModel\GroupRepository;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Quote\Model\Quote\Item;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\Store;
+use Magento\Tax\Model\Calculation\Proxy;
+use Magento\Tax\Model\Config;
+
+class ShippingTaxPlugin
+{
+    public const CONFIG_PATH_DYNAMIC_SHIPPING_TAX_CLASS = 'tax/classes/dynamic_shipping_tax_class';
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var Cart
+     */
+    private $cart;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * @var GroupRepository
+     */
+    private $groupRepository;
+
+    /**
+     * @var Proxy
+     */
+    private $taxCalculation;
+
+    /**
+     *
+     * @param ScopeConfigInterface $scopeConfig
+     * @param Cart                 $cart
+     * @param Session              $customerSession
+     * @param GroupRepository      $groupRepository
+     * @param Proxy                $taxCalculation
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        Cart $cart,
+        Session $customerSession,
+        GroupRepository $groupRepository,
+        Proxy $taxCalculation
```

Comment:
We should not depend on a proxy.

Reply:
The Plugin does not have a around method. I think the plugin will not work in general.